### PR TITLE
updater for a resinOS variant sometimes (rarely) found in production

### PR DESCRIPTION
**Pull request by imrehg**
* Fixing the state partition location and label.
* Use correct slug and sidestep sanity check as we'd know the results

Change-type: patch

### CODE REVIEW

File: upgrade-2.x.sh [hunk 1]
```diff
@@ -623,6 +623,22 @@ function find_partitions {
...
+# Globals:
+#   root_dev
+#   state_part
+# Returns:
+#   None
+#######################################
+function fix_state_label() {
+    if [ "${state_part}" != "/mnt/state" ]; then
+        log "Need to apply proper state partition label."
+        e2label ${root_dev}5 "resin-state"
```
Comment (agherzan) [inline upgrade-2.x.sh]: Do we have boards with different partition indexes? O remember tx2 had something like that. @telphan ? @floion ?

Comment (floion) [reply upgrade-2.x.sh]: Yes @agherzan @imrehg 
The edison has a different layout and the 5th partition is not the state partition
And @telphan can chime in for the TX2 / TX1 (actually should chime in)

Comment (telphan) [reply upgrade-2.x.sh]: Yes, state partition is different in TX1 and TX2, I think it is 14 and 16 respectively.

Comment (agherzan) [reply upgrade-2.x.sh]: Thanks guys.

Comment (imrehg) [reply upgrade-2.x.sh]: @agherzan I have a better way to handle this in mind, will try it out and rewrite if that works.

**Comment by imrehg** [comment]
@agherzan thanks! I'm checking with our solution architects, to see whether we'll need to merge this at all, or use the code separately (since it's only for one customer). But the feedback is very helpful so far!

**Comment by imrehg** [comment]
@agherzan redone the state partition check, let me know if this is any better?
Also added everything that's needed to be able to update devices of this kind without any extra flags.
This kept as a single commit, as later we might remove these changes, if no longer needed.

**Comment by xginn8** [comment]
closing for now, can reopen if necessary

**Pull request closed without merging**